windows/src/engine/keyman32/K32_load.cpp

```diff
@@ -75,12 +75,55 @@ BOOL GetKeyboardFileName(LPSTR kbname, LPSTR buf, int nbuf)
   return n;
 }
 
+BOOL LoadlpKeyboardCore(int i)
+{
+  SendDebugMessageFormat(0, sdmAIDefault, 0, "LoadlpKeyboardCore: Enter ---");
+  PKEYMAN64THREADDATA _td = ThreadGlobals();
+  if (!_td) return FALSE;
+  if (_td->lpKeyboards[i].coreKeyboard) return TRUE;
+  if (_td->lpActiveKeyboard == &_td->lpKeyboards[i]) _td->lpActiveKeyboard = NULL;  // I822 TSF not working
+
+  if (_td->lpKeyboards[i].lpActiveKBState) {
+    km_kbp_state_dispose(_td->lpKeyboards[i].lpActiveKBState);
```

Comment:
Is this an unlikely state to be in, given line 83 above? Should we be logging this as a 'hey how did we get here?'?

Reply:
So this highlights a problem in my releasing memory code. The windows engine wants to treat a core keyboard state and a core keyboard as a 1-1 relationship. (Core does allow for multiple states for the one keyboard).  With that in mind we need to create a `ReleaseKeyboardMemoryStateCore` that is called in `ReleaseKeyboards`.  Then line 86 as you have said should not happen.